src/GafferOSL/OSLImage.cpp

```diff
@@ -148,6 +182,20 @@ void OSLImage::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outpu
 		outputs.push_back( outPlug()->channelNamesPlug() );
 		outputs.push_back( outPlug()->channelDataPlug()	);
 	}
+	else if( input == inPlug()->formatPlug() )
```

Comment:
I'm planning on refactoring dirty propagation so that ImageNodes have `virtual bool affectsFormat()` etc methods. In the run up to that I'm trying to ensure that all `affects()` implementations are structured with a single conditional per output plug, rather than having a single conditional drive multiple outputs. Although this can be wordier, I'm finding that it is far easier to verify the correctness of the code this way - for each hash/compute, you just need to check that all the inputs used are listed in the one relevant conditional. Could you take that approach here please?